business-registry-dashboard/app/pages/incorporateNow.vue

```diff
@@ -72,34 +70,41 @@ onMounted(async () => {
       }
     }
 
-    // Load user's affiliated businesses and check if this name request is already affiliated
-    // This prevents unauthorized access to name requests that belong to other users
+    // Load and check if the NR is already affiliated
     await affStore.loadAffiliations()
     const isAffiliated = isNameRequestAffiliated(affStore.affiliations.results, business.nameRequest?.nrNumber)
 
-    // Verify business eligibility and proceed with incorporation
-    if (business.nameRequest.requestActionCd === NrRequestActionCodes.NEW_BUSINESS &&
-        ELIGIBLE_CORP_TYPES.includes(business.nameRequest.legalType) && isAffiliated) {
-      try {
-        // Create the business and redirect to dashboard
-        const identifier = await createBusinessFromNR(business)
-        await affNav.goToDashboard(identifier)
-      } catch (error) {
-        console.error('Error creating named business:', error)
-        brdModal.openMagicLinkModal(
-          t('error.magicLinkCreateBusinessFailed.title'),
-          t('error.magicLinkCreateBusinessFailed.description')
-        )
-      }
+    // Create affiliation if it doesn't exist
+    if (!isAffiliated) {
+      await affStore.createNRAffiliation({
+        businessIdentifier: business.nameRequest?.nrNumber,
+        ...(phone && { phone }),
+        ...(email && { email })
+      })
+    }
+
+    // Get the temporary business identifier if it exists
+    const tempBusinessIdentifier = getTempBusinessIdentifierOfNameRequest(affStore.affiliations.results, business.nameRequest?.nrNumber)
```

Comment:
I noticed a bug. If the user clicks on the link again and again, a new draft is created every time. This now, prevents that. IF a draft IA has been created before, we will direct the user to the previously created draft temp business identifier.

Reply:
Great idea.

Future: check the Auth db whether any business exists for this IA (across the board). This would prevent another account from affiliating the NR and creating a second IA from it.

Reply:
Agreed. New ticket: https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/25005